okteto init fails with a nullref if the deployment doesn't have pods

**Describe the bug**


**To Reproduce**
Steps to reproduce the behavior:
1. create a kubernetes deployment and scale it to 0 (or put a bad image so a pod is not created)
2. run `okteto init`
3. See error

```
 okteto init -n=openfaas-fn -l=debug
This command walks you through creating an okteto manifest.
It only covers the most common items, and tries to guess sensible defaults.
See https://okteto.com/docs/reference/manifest for the official documentation about the okteto manifest.
DEBU[0000] Language guesses:
 100.00%	TypeScript
INFO[0000] language 'javascript' inferred for your current directory
INFO[0000] autogenerated name: openfaas-webassembly
 ✓  hello
⠋ Analyzing deployment 'hello'... INFO[0001] replicaset hello-7d7bbd78d8 with revison 3 is progressing
INFO[0001] Analysis for deployment 'hello' failed: no pod is running for deployment 'hello'
Analysis for deployment 'hello' failed: no pod is running for deployment 'hello'
DEBU[0001] checking persistent volumes support in your cluster...
DEBU[0001] found default storage class 'standard'
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x25e873b]

goroutine 1 [running]:
github.com/okteto/okteto/cmd/init.Run(0x7ffeefbff56f, 0xb, 0x2e6a2e3, 0xa, 0x0, 0x0, 0xc00005c034, 0x29, 0xc00119fc00, 0x1357785, ...)
	/home/circleci/project/cmd/init/init.go:132 +0x54b
github.com/okteto/okteto/cmd/init.Init.func1(0xc00127e780, 0xc001219220, 0x0, 0x2, 0x0, 0x0)
	/home/circleci/project/cmd/init/init.go:60 +0xf8
github.com/spf13/cobra.(*Command).execute(0xc00127e780, 0xc001219200, 0x2, 0x2, 0xc00127e780, 0xc001219200)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:826 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0xc001278500, 0xc00119ff58, 0x1, 0x1)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
main.main()
	/home/circleci/project/main.go:90 +0x84d
```

**Desktop (please complete the following information):**
 - OS: macOS
 - Version 1.8.13
 - repo is in typescript

--- a/cmd/init/init.go
+++ b/cmd/init/init.go
@@ -119,12 +119,13 @@ func Run(namespace, devPath, language, workDir string, overwrite bool) error {
 			suffix := fmt.Sprintf("Analyzing deployment '%s'...", d.Name)
 			spinner := utils.NewSpinner(suffix)
 			spinner.Start()
-			dev, err = initCMD.SetDevDefaultsFromDeployment(dev, d, container)
+			err = initCMD.SetDevDefaultsFromDeployment(dev, d, container)
 			spinner.Stop()
 			if err == nil {
 				log.Success(fmt.Sprintf("Deployment '%s' successfully analyzed", d.Name))
 			} else {
 				log.Yellow(fmt.Sprintf("Analysis for deployment '%s' failed: %s", d.Name, err))
+				linguist.SetForwardDefaults(dev, language)
 			}
 		}
 

--- a/pkg/cmd/init/run.go
+++ b/pkg/cmd/init/run.go
@@ -34,28 +34,22 @@ var (
 )
 
 //SetDevDefaultsFromDeployment sets dev defaults from a running deployment
-func SetDevDefaultsFromDeployment(dev *model.Dev, d *appsv1.Deployment, container string) (*model.Dev, error) {
+func SetDevDefaultsFromDeployment(dev *model.Dev, d *appsv1.Deployment, container string) error {
 	ctx := context.Background()
 	c, _, _, err := k8Client.GetLocal()
 	if err != nil {
-		return nil, err
+		return err
 	}
+
+	setAnnotationsFromDeployment(dev, d)
+	setNameAndLabelsFromDeployment(ctx, dev, d)
 
 	pod, err := getRunningPod(d, container, c)
 	if err != nil {
-		return nil, err
+		return err
 	}
-
-	dev = setNameAndLabelsFromDeployment(ctx, dev, d)
-	dev = setAnnotationsFromDeployment(dev, d)
-	dev = setResourcesFromPod(dev, pod, container)
-
-	dev, err = setForwardsFromPod(ctx, dev, pod, c)
-	if err != nil {
-		return nil, err
-	}
-
-	return dev, nil
+	setResourcesFromPod(dev, pod, container)
+	return setForwardsFromPod(ctx, dev, pod, c)
 }
 
 func getRunningPod(d *appsv1.Deployment, container string, c *kubernetes.Clientset) (*apiv1.Pod, error) {
@@ -84,10 +78,10 @@ func getRunningPod(d *appsv1.Deployment, container string, c *kubernetes.Clients
 	return pod, nil
 }
 
-func setForwardsFromPod(ctx context.Context, dev *model.Dev, pod *apiv1.Pod, c *kubernetes.Clientset) (*model.Dev, error) {
+func setForwardsFromPod(ctx context.Context, dev *model.Dev, pod *apiv1.Pod, c *kubernetes.Clientset) error {
 	ports, err := services.GetPortsByPod(pod, c)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	seenPorts := map[int]bool{}
 	for _, f := range dev.Forward {
@@ -110,10 +104,10 @@ func setForwardsFromPod(ctx context.Context, dev *model.Dev, pod *apiv1.Pod, c *
 			},
 		)
 	}
-	return dev, nil
+	return nil
 }
 
-func setNameAndLabelsFromDeployment(ctx context.Context, dev *model.Dev, d *appsv1.Deployment) *model.Dev {
+func setNameAndLabelsFromDeployment(ctx context.Context, dev *model.Dev, d *appsv1.Deployment) {
 	for _, l := range componentLabels {
 		component := d.Labels[l]
 		if component == "" {
@@ -121,21 +115,18 @@ func setNameAndLabelsFromDeployment(ctx context.Context, dev *model.Dev, d *apps
 		}
 		dev.Name = component
 		dev.Labels = map[string]string{l: component}
-		return dev
+		return
 	}
 	dev.Name = d.Name
-	return dev
-
 }
 
-func setAnnotationsFromDeployment(dev *model.Dev, d *appsv1.Deployment) *model.Dev {
+func setAnnotationsFromDeployment(dev *model.Dev, d *appsv1.Deployment) {
 	if v := d.Annotations[okLabels.FluxAnnotation]; v != "" {
 		dev.Annotations = map[string]string{"fluxcd.io/ignore": "true"}
 	}
-	return dev
 }
 
-func setResourcesFromPod(dev *model.Dev, pod *apiv1.Pod, container string) *model.Dev {
+func setResourcesFromPod(dev *model.Dev, pod *apiv1.Pod, container string) {
 	for i := range pod.Spec.Containers {
 		if pod.Spec.Containers[i].Name != container {
 			continue
@@ -148,7 +139,6 @@ func setResourcesFromPod(dev *model.Dev, pod *apiv1.Pod, container string) *mode
 				},
 			}
 		}
-		break
+		return
 	}
-	return dev
 }

